catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardImpl.java

```diff
@@ -710,7 +710,15 @@ public Attribute getAttribute(String name) {
      *            the value of the {@link Attribute}
      */
     public void setAttribute(String name, Serializable value) {
-        setAttribute(new AttributeImpl(name, value));
+        if (value != null && name.equalsIgnoreCase(RESOURCE_URI)
```

Comment:
you could extend this to the generic case by checking against all the `AttributeDescriptors` of the metacard type. 
`if (value != null && this.type.getAttributeDescriptor(name).getBinding().class.isAssignableFrom(value.getClass())))`
i think. something like that at least.